### PR TITLE
chore: update hermetic build workflow

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Secure generation workflow (use environment variable to avoid script injection), inspired by https://github.com/googleapis/java-bigtable/pull/2317
- Generalize script so that it can run both in google-cloud-java and handwritten libraries.

This workflow will serve as a source of true for generation workflows in handwritten libraries.

### CODE REVIEW

**Review by diegomarquezp — APPROVED** [review]
LGTM, thanks!

File: .github/scripts/hermetic_library_generation.sh [hunk 1]
```diff
@@ -54,22 +47,18 @@ shift
...
 
 if [ -z "${current_branch}" ]; then
-  echo "missing required argument --current_branch"
+  echo "Missing required argument --current_branch"
   exit 1
 fi
 
 if [ -z "${generation_config}" ]; then
   generation_config=generation_config.yaml
-  echo "Use default generation config: ${generation_config}"
```
Comment (blakeli0) [inline .github/scripts/hermetic_library_generation.sh:68]: Can we keep the image tag as a parameter? In long term, we do want to pass it in and remove the generator version from the generation_config.yaml.

File: .github/scripts/hermetic_library_generation.sh [hunk 2]
```diff
@@ -101,8 +94,15 @@ docker run \
   gcr.io/cloud-devrel-public-resources/java-library-generation:"${image_tag}" \
   --baseline-generation-config-path="${workspace_name}/${baseline_generation_config}" \
   --current-generation-config-path="${workspace_name}/${generation_config}"
+
 # commit the change to the pull request.
-git add java-* pom.xml gapic-libraries-bom/pom.xml versions.txt
+if [[ $(basename "$(pwd)") == "google-cloud-java" ]]; then
```
Comment (blakeli0) [inline .github/scripts/hermetic_library_generation.sh:99]: Do we have to differentiate between google-cloud-java and java-* repos? Can we just do `git add .`?

File: .github/scripts/hermetic_library_generation.sh (hunk 1 above)
Comment (JoeWang1127) [reply .github/scripts/hermetic_library_generation.sh:68]: Sorry I missed this comment. I'll create a follow up.

**Pull request merged**